coding-standards/log-format-mobile-app.md

```diff
@@ -0,0 +1,53 @@
+# Log format
+
+## Requirements
+To create a common log format that fits the most mobile technologies and aids in debugging:
+
+- Include the severity of the log to understand its importance.
+- Include the type of log to segregate between info, error, or warning.
+- Include screen or controller where applicable.
+- Include a unique ID to trace requests across different components or screens.
+- Include source context, which could be a function name, module, or screen name.
+- Include relevant data in JSON format, such as user information, activity details, performance metrics, and data changes.
+- Facilitate debugging of production issues by providing sufficient information in logs, such as stack traces, error messages, and contextual details.
+
+> **Warning**
+> Going against GDPR regulations is a absolute NO
+
+## Proposed solution
+
+``` json
+ {"timestamp": "[timestamp]","level": "[level]","severity": "[severity]","tracebackId": "[unique id]","screenName": "[screen or controller name]","source": "[function/module name]","data": [any relevant data of type JSON],"message": "[log message]","stackTrace": "[stackTrace]"}
+```
+
+### Explanation of each component: 
+
+| Component     | Description                                     |
+| ------------- | ----------------------------------------------- |
+| [timestamp]   | Represents the timestamp when the log entry was created. Use ISO 8601 format, which is widely recognized and provides a standardized representation of date and time (Mandatory) |
+| [level]       | Represents the log level, such as INFO, WARN, ERROR, or FATAL (Mandatory) |
+| [severity]    | Represents the severity level associated with the log entry (e.g., LOW, MEDIUM, HIGH) (Mandatory) |
+| [tracebackId] | A unique identifier associated with the request (Mandatory) |
+| [screenName]  | Name of the screen or controller where the log originated (Optional) |
+| [httpMethod]  | Type of request (Optional) |
+| [source]      | Function name, Module & Sub module names etc. (Mandatory) |
+| [data]        | Any relevant data like user information, activity, performance metrics, data changes etc. (Optional) |
+| [message]     | Contains the log message or description of the event (Mandatory) |
+| [stackTrace]  | The stack trace of the exception, providing details about the method calls leading to the exception. It includes file names, line numbers, and method names (Optional) |
```

Comment:
Right! It may need legitimate information and user consent. 
We can ignore this.

Reply:
The session id and other information related to device or browser we can have them in data as:
``` json
{"data": {"userId": "123", "deviceInfo": {"platform": "iOS", "version": "14.5"}}}
{"data": {"userId": "123", "deviceInfo": {"platform": "chrome", "version": "14.5"}}}
```

Reply:
You need to make it a point in the document so that users know that they have to always sent this info.

But I still think these kind of things which repeat for every log we can have it in main property instead of data.

Since data will be more like if I have a exception in a function then I will log the function arguments are data. To know with what data the function was called.

Reply:
I suggest we keep device info in a different property since this will be there always for every log.

Reply:
Agree with keeping device info in a different property.

Reply:
Agree! Will make changes